tests/Tokenizer/Analyzer/ClassAnalyzerTest.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tests\TestCase;
+use PhpCsFixer\Tokenizer\Analyzer\ClassAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Pol Dellaiera <pol.dellaiera@protonmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Tokenizer\Analyzer\ClassAnalyzer
+ */
+final class ClassAnalyzerTest extends TestCase
+{
+    /**
+     * @param string $code
+     * @param bool   $hasExtends
+     * @param mixed  $classIndex
+     *
+     * @dataProvider provideClassHasExtendsCases
+     */
+    public function testClassHasExtends($classIndex, $code, $hasExtends)
```

Comment:
I'd go either with testing here result of `getClassExtends` (so replacing `$hasExtends` with `null` or `TypeAnalysis` expectations) or add `hasClassExtends` (`isClassExtendsAnything`?) which would basically check if `getClassExtends` returns `null` or not.

Well - on the 2nd thought - I'd expect here we have direct tests for `getClassExtends` anyway.

Reply:
Okay, fixed now.